lib/utility/es_client.rb

```diff
@@ -11,6 +11,10 @@
 
 module Utility
   class EsClient < Elasticsearch::Client
+    def self.instance
```

Comment:
This is not really a singleton. You can still call `Utility::EsClient.new`

Reply:
Good point, I actually reverted the code inside `es_client.rb` to be a real singleton - before as you noticed it was allowing you to work with class instances OR singleton, which is a really really confusing way to work with a class.

Reply:
If we want to make `EsClient` a singleton I would do it with this https://ruby-doc.org/stdlib-2.6/libdoc/singleton/rdoc/Singleton.html. It will simplify the code a bit and also make sure that `initialize` method is private.

Reply:
TIL there's a Singleton module 👍